src/gpu/metal/SDL_gpu_metal.m

```diff
@@ -1121,25 +1121,41 @@ static void METAL_ReleaseGraphicsPipeline(
 
         if (createinfo->target_info.has_depth_stencil_target) {
             pipelineDescriptor.depthAttachmentPixelFormat = SDLToMetal_TextureFormat(createinfo->target_info.depth_stencil_format);
+            pipelineDescriptor.stencilAttachmentPixelFormat = SDLToMetal_TextureFormat(createinfo->target_info.depth_stencil_format);
```

Comment:
If the format doesn't have stencil, then the validation layer will assert here. So we need to check `if (IsStencilFormat(createinfo->target_info.depth_stencil_format))` before assigning `stencilAttachmentPixelFormat`.

Reply:
Sent a repo invite, easy to commit but I'm not able to test atm